runtime/lua/vim/lsp/protocol.lua

```diff
@@ -923,4 +923,274 @@ function protocol.resolve_capabilities(server_capabilities)
+  --- is of type {@link CompletionItem CompletionItem[]} or {@link CompletionList}
+  --- or a Thenable that resolves to such.
+  --- The request can delay the computation of the {@link CompletionItem.detail `detail`}
+  --- and {@link CompletionItem.documentation `documentation`} properties to the `completionItem/resolve`
+  --- request. However, properties that are needed for the initial sorting and filtering, like `sortText`,
+  --- `filterText`, `insertText`, and `textEdit`, must not be changed during resolve.
+  textDocument_completion = 'textDocument/completion',
+  --- Request to request hover information at a given text document position. The request's
+  --- parameter is of type {@link TextDocumentPosition} the response is of
+  --- type {@link Hover} or a Thenable that resolves to such.
+  textDocument_hover = 'textDocument/hover',
+  textDocument_signatureHelp = 'textDocument/signatureHelp',
+  --- A request to resolve the definition location of a symbol at a given text
+  --- document position. The request's parameter is of type [TextDocumentPosition]
+  --- (#TextDocumentPosition) the response is of either type {@link Definition}
+  --- or a typed array of {@link DefinitionLink} or a Thenable that resolves
+  --- to such.
+  textDocument_definition = 'textDocument/definition',
+  --- A request to resolve project-wide references for the symbol denoted
+  --- by the given text document position. The request's parameter is of
+  --- type {@link ReferenceParams} the response is of type
+  --- {@link Location Location[]} or a Thenable that resolves to such.
+  textDocument_references = 'textDocument/references',
+  --- Request to resolve a {@link DocumentHighlight} for a given
+  --- text document position. The request's parameter is of type [TextDocumentPosition]
+  --- (#TextDocumentPosition) the request response is of type [DocumentHighlight[]]
+  --- (#DocumentHighlight) or a Thenable that resolves to such.
+  textDocument_documentHighlight = 'textDocument/documentHighlight',
+  --- A request to list all symbols found in a given text document. The request's
+  --- parameter is of type {@link TextDocumentIdentifier} the
+  --- response is of type {@link SymbolInformation SymbolInformation[]} or a Thenable
+  --- that resolves to such.
+  textDocument_documentSymbol = 'textDocument/documentSymbol',
+  --- A request to provide commands for the given text document and range.
+  textDocument_codeAction = 'textDocument/codeAction',
+  --- A request to provide code lens for the given text document.
+  textDocument_codeLens = 'textDocument/codeLens',
+  --- A request to provide document links
+  textDocument_documentLink = 'textDocument/documentLink',
+  --- A request to to format a whole document.
+  textDocument_formatting = 'textDocument/formatting',
+  --- A request to format a range in a document.
+  textDocument_rangeFormatting = 'textDocument/rangeFormatting',
+  --- A request to format ranges in a document.
+  --- @since 3.18.0
```

Comment:
I think Nvim LSP client is currently targeting 3.17, at least that's what lsp_types.lua last used.

Let's stick with 3.17 and discuss updating to a newer version in a future PR.